cql/src/main/java/org/apache/cassandra/stargate/transport/internal/HAProxyProtocolDetectingDecoder.java

```diff
@@ -87,4 +74,27 @@ private static boolean match(byte[] prefix, ByteBuf buffer, int idx) {
     }
     return true;
   }
+
+  static class InboundHAProxyHandler extends SimpleChannelInboundHandler<HAProxyMessage> {
```

Comment:
Extracted into separate class to create a stateless singleton, to avoid allocating instance per call. Not a big deal but seemed pointless to create throw-away instances. Main change is the validation.